### ISSUE TITLE
Allow urls to not be forcefully opened in the PWA

### ISSUE THREAD

**Issue by PieterjanDeClippel** [WICG/pwa-url-handler#28]
I've built an angular PWA where the user is able to sign-in using Microsoft, Google, Facebook, Twitter. When the user clicks a button, I'm using `window.open('https://example.com/signin-microsoft')` to show the Microsoft login page. This is in order not to interrupt the web application.

The problem is, when the user is navigating the angular app in Chrome mobile while the PWA is installed, `window.open()` opens the PWA instead of opening the url in a new tab. This causes the login-callback-page not to be able to message back the result of the login action.

I used to get around this by creating a subdomain hosting the endpoints for my external login actions. But now since I've implemented 2FA, there's no way to have 2FA working when the `ExternalLoginSigninAsync` and `TwoFactorAuthenticatorSigninAsync` are hosted at different domains ('external.example.com' vs 'example.com').

The bottom line is: a developer MUST BE allowed to have `window.open` open the url in a new window and prevent it from opening the PWA.

For more information about what's already been tried and errored: [consult this link](https://stackoverflow.com/questions/68518730/pwa-prevent-urls-from-opening-in-the-pwa?noredirect=1#comment121092911_68518730)

**Comment by LuHuangMSFT** [comment]
Hello @PieterjanDeClippel. That's a valid and common scenario. 

A couple of quick questions first: 1.) are you using the "url_handlers" member in your PWA manifest? 2.) Are you describing behavior on Android? 

If the answers are "no", and "yes", this might be an intent filter/deep linking issue with Android PWAs. 

PWA URL handling as described in the explainer [here](https://github.com/WICG/pwa-url-handler/blob/main/explainer.md) is implemented in Chromium for Windows, Mac OS, and Linux (desktop). It is not currently implemented for mobile platforms like Android or iOS. PWA URL handling also allows you, the developer, to exclude paths or URLs from URL handling behavior in a configuration file.

**Comment by LuHuangMSFT** [comment]
I suspect this is an Android WebAPK behavior.

> When a Progressive Web App is installed on Android, it will register a set of intent filters for all URLs within the scope of the app. When a user clicks on a link that is within the scope of the app, the app will be opened, rather than opening within a browser tab.

https://developers.google.com/web/fundamentals/integration/webapks#android_intent_filters

**Comment by PieterjanDeClippel** [comment]
Hi @LuHuangMSFT,

The answers are as you stated:
- Are you using the "url_handlers" member in your PWA manifest? **No**
- Are you describing behavior on Android? **Yes**

To clarify, the PWA <-> website on Windows are behaving totally fine.
The behavior only occurs on android.

**Comment by PieterjanDeClippel** [comment]
I did a little test, and installed the PWA for all of Chrome, Edge and Firefox (only on android, since the problem is only on android).
The `window.open` behavior is as follows:
- **Chrome:** Opens in the chrome-installed-PWA
- **Edge:** Opens in the edge-browser
- **Firefox:** Opens in the firefox-browser

**Comment by LuHuangMSFT** [comment]
As far as I know, only WebAPKs installed from Chrome have the OS integration necessary for the suspected intent handling behavior. Most notably, only Chrome-installed PWAs are visible in the system app drawer. None Chrome-installed PWAs appear as home screen shortcuts. I'm not an expert on this topic so I cannot help much more than that.

> On a second take in Edge, when calling window.open I do get a picker menu to choose from either the chrome PWA, and Chrome. Not Edge.

I believe this a bug and I'll file it with the Edge team for investigation. Could you tell me Android Edge version you used and the Android OS version? 

I believe it is possible to exclude paths from the intent filters registered by the WebAPK, although I'm not sure how that's done with angular/pwa. If you can do that, it would definitively root cause the problematic behavior and make it go away. 

I will close this issue now as we were able to confirm that the "url_handlers" PWA manifest field was not used.

**Comment by LuHuangMSFT** [comment]
> > On a second take in Edge, when calling window.open I do get a picker menu to choose from either the chrome PWA, and Chrome. Not Edge.

> I believe this a bug and I'll file it with the Edge team for investigation. Could you tell me Android Edge version you used and the Android OS version?

After more thought, I think this is correct Android behavior if you had both the Edge and Chrome PWAs installed. The OS intent handling system "saw" the link activation from `window.open` and showed the available intent handlers (the Chrome-installed PWA with intent filters, and the OS default browser. Can you confirm that Chrome was the OS default browser at that time?

**Comment by PieterjanDeClippel** [comment]
> > > On a second take in Edge, when calling window.open I do get a picker menu to choose from either the chrome PWA, and Chrome. Not Edge.
> 
> > I believe this a bug and I'll file it with the Edge team for investigation. Could you tell me Android Edge version you used and the Android OS version?
> 
> After more thought, I think this is correct Android behavior if you had both the Edge and Chrome PWAs installed. The OS intent handling system "saw" the link activation from `window.open` in Edge and showed the available intent handlers it knows about (the Chrome-installed PWA with intent filters, and the OS default browser. Can you confirm that Chrome was the OS default browser at that time?

Yeah right. I can confirm that Chrome is the default browser AND the PWA was atm installed through Chrome

**Comment by PieterjanDeClippel** [comment]
Conditions:
- On Android
- Edge is the default browser
- Chrome PWA is installed
- Edge PWA is "installed"

Result:
- Through Edge webbrowser
- Call `window.open`
- window is opened in Edge tab
- Everything works fine

Conditions:
- On Android
- Chrome is the default browser
- Chrome PWA is installed
- Edge PWA is "installed"

Result:
- Through the Edge webbrowser
- Call `window.open`
- Strangely the window is opened in the Edge webbrowser now...

**Comment by PieterjanDeClippel** [comment]
In another test:
- Conditions
    - On Android
    - Edge is default browser
    - Both the Chrome PWA and Edge PWA are installed
- Result
    - When opening Edge and calling `window.open`, the url is opened in the Chrome PWA
